modules/internal/CPtr.chpl

```diff
@@ -223,8 +223,26 @@ module CPtr {
     :arg arr: the array for which we should retrieve a pointer
     :returns: a pointer to the array data
   */
-  inline proc c_ptrTo(arr: []) where isRectangularArr(arr) {
-    if !arr._value.oneDData then halt("error: c_ptrTo(multi_ddata array");
+  inline proc c_ptrTo(arr: []) where isRectangularArr(arr) && !arr._value.isDefaultRectangular() {
+    if !chpl__getActualArray(arr).oneDData then halt("error: c_ptrTo(multi_ddata array");
+    return c_pointer_return(arr[arr.domain.low]);
+  }
+
+  pragma "no doc"
+  inline proc c_ptrTo(arr: []) where arr._value.isDefaultRectangular() {
+    const val = arr._value;
+    if chpl__isArrayView(val) {
```

Comment:
For the CPtr -> domain map refactoring, it doesn't matter to me deeply whether that's part of this PR or a follow-on.

For making isDefaultRectangular not return true for views, I'd do that as a separate PR (though whether it goes in before or after this one doesn't matter to me).  I'm thinking I probably should've noticed this issue earlier, but somehow didn't...